hack/generator/pkg/astmodel/enum_type.go

```diff
@@ -29,25 +30,98 @@ func NewEnumType(baseType *PrimitiveType, options []EnumValue) *EnumType {
 		return options[left].Identifier < options[right].Identifier
 	})
 
-	return &EnumType{BaseType: baseType, options: options}
+	return &EnumType{baseType: baseType, options: options}
+}
+
+func (enum *EnumType) AsDeclarations(codeGenerationContext *CodeGenerationContext, name *TypeName, description *string) []ast.Decl {
+	var specs []ast.Spec
+	for _, v := range enum.options {
+		s := enum.createValueDeclaration(v, name)
+		specs = append(specs, s)
+	}
+
+	declaration := &ast.GenDecl{
+		Tok:   token.CONST,
+		Doc:   &ast.CommentGroup{},
+		Specs: specs,
+	}
+
+	result := []ast.Decl{
+		enum.createBaseDeclaration(codeGenerationContext, name, description),
+		declaration}
+
+	return result
+}
+
+func (enum *EnumType) createBaseDeclaration(codeGenerationContext *CodeGenerationContext, name *TypeName, description *string) ast.Decl {
+	identifier := ast.NewIdent(name.Name())
+
+	typeSpecification := &ast.TypeSpec{
+		Name: identifier,
+		Type: enum.baseType.AsType(codeGenerationContext),
+	}
+
+	declaration := &ast.GenDecl{
+		Tok: token.TYPE,
+		Doc: &ast.CommentGroup{},
+		Specs: []ast.Spec{
+			typeSpecification,
+		},
+	}
+
+	if description != nil {
+		declaration.Doc.List = append(
+			declaration.Doc.List,
+			&ast.Comment{Text: "\n/*" + *description + "*/"})
+	}
+
+	validationComment := GenerateKubebuilderComment(enum.CreateValidation())
+	declaration.Doc.List = append(
+		declaration.Doc.List,
+		&ast.Comment{Text: "\n" + validationComment})
+
+	return declaration
+}
+
+func (enum *EnumType) createValueDeclaration(value EnumValue, name *TypeName) ast.Spec {
```

Comment:
I think these parameters should be the other way around ... maybe.

Reply:
make s sense